provider/marathon/marathon_test.go

```diff
@@ -1801,3 +1807,33 @@ func TestParseIndex(t *testing.T) {
 		})
 	}
 }
+
+func TestMarathonGetBasicAuth(t *testing.T) {
+	provider := &Provider{}
+
+	applications := []struct {
+		application marathon.Application
+		expected    []string
+	}{
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{}},
+			expected: []string{},
+		},
+		{
+			application: marathon.Application{
+				Labels: &map[string]string{
+					"traefik.frontend.auth.basic": "user:password",
+				},
+			},
+			expected: []string{"user:password"},
+		},
+	}
+
+	for _, a := range applications {
```

Comment:
Sure thing: [Here you go](https://github.com/timoreimann/traefik/blob/bdad3d2f281126d58206fcba86c2862133850881/provider/marathon/marathon_test.go#L840-L849) (from an in-flight PR of mine).

Key points:

1. Capture the loop variable like `c := c`.
1. Pass an expressive description/test name as first parameter to `t.Run()`.
1. Call `t.Parallel()` first thing inside the body of `t.Run()`.

If you're interested in a bit of background, I suggest [this blog post](https://blog.golang.org/subtests).

Let me know if you have further questions!